packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1253,19 +1276,22 @@ export class EthModule {
       if (trace.bytecode === undefined) {
         this._logger.logWithTitle(
           "Contract deployment",
-          UNRECOGNIZED_CONTRACT_NAME
+          UNRECOGNIZED_CONTRACT_NAME,
+          indent
         );
       } else {
         this._logger.logWithTitle(
           "Contract deployment",
-          trace.bytecode.contract.name
+          trace.bytecode.contract.name,
+          indent
         );
       }
 
       if (trace.deployedContract !== undefined && trace.error === undefined) {
         this._logger.logWithTitle(
           "Contract address",
-          bufferToHex(trace.deployedContract)
+          bufferToHex(trace.deployedContract),
+          indent
         );
       }
```

Comment:
Instead of passing down `indent` as a parameter everywhere, I would add a `setIndent` method to the `ModulesLogger` class. It would make all the subsequent logs added with `log` and `logWithTitle` methods indented. At the end of indentation the block you'd call `setIndent(0)`.